### PR TITLE
macOS app: improve rough edges

### LINKED ISSUE #745: onboarding notif page on macos
macos native notif prompt bad, make gif or something to make it more explicit what to do

image

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
- Show a video accepting notifs on macOS app on Allow Notifications page: closes #745 
- Add a settings redirect like contacts to notifs and camera access flow

Tested notifs page with prod Mac app but didn't record video and now the app is the one from master, here's simulator video of camera change:

https://github.com/daimo-eth/daimo/assets/6984346/ffd535fe-05fd-4d2e-9f13-2f3b8e194fbd

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/SettingsScreen.tsx [hunk 1]
```diff
@@ -325,15 +324,7 @@ function PendingDeviceRow({ slot }: { slot: number }) {
 }
 
 function DetailsSection({ account }: { account: Account }) {
-  const enableNotifications = async () => {
-    await Notifications.requestPermissionsAsync();
-    try {
-      getPushNotificationManager().savePushTokenForAccount();
```
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/view/screen/SettingsScreen.tsx:331]: @dcposch could you double check these notifs changes don't cause any edge case regression? I was a bit confused why this one didn't go through the extra logic for android in `maybeSavePushTokenForAccount`

**Comment by nalinbhardwaj** [comment]
@koredefashokun [last commit](https://github.com/daimo-eth/daimo/pull/826/commits/077c43bfd132fd7b1ef0a5a8627e359271d7aecf) is a fix for the image glitching on re-render and flashing the default pic:

https://github.com/daimo-eth/daimo/assets/6984346/6ed41462-2e74-41ac-9d85-5a983a3d2703

Lmk if you see a cleaner way to fix that, the change works but not the cleanest.

File: apps/daimo-mobile/src/view/screen/onboarding/AllowNotificationsPage.tsx [hunk 2]
```diff
@@ -31,24 +38,61 @@ export function AllowNotificationsPage({ onNext }: { onNext: () => void }) {
...
+            style={{
+              backgroundColor: color.primary,
+              alignSelf: "center",
+              borderRadius: 8,
+              // Hardcoded from actual video size, prevents glitchy jump on load
+              width: 377,
+              height: 145,
+            }}
+            source={{
+              uri: `${daimoDomainAddress}/assets/macos-notifs-explainer.mp4`,
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/onboarding/AllowNotificationsPage.tsx:53]: 👀

File: apps/daimo-mobile/src/logic/notify.ts [hunk 3]
```diff
@@ -125,3 +117,60 @@ class PushNotificationManager {
...
+      undefined,
+      [
+        {
+          text: `Continue`,
+        },
+      ],
+      { cancelable: false }
+    );
+    return;
+  }
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/notify.ts:170]: this section might not be necessary

File: apps/daimo-mobile/src/logic/scanner.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { useEffect, useState } from "react";
+
+import { askOpenSettings } from "./settings";
+
+interface ScannerAccess {
+  permission: BarCodeScanner.PermissionResponse | undefined;
+  ask: () => Promise<void>;
+}
+
+export function useScannerAccess(): ScannerAccess {
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/scanner.ts:11]: nice cleanup

File: apps/daimo-mobile/src/view/screen/SettingsScreen.tsx (hunk 1 above)
Comment (dcposch) [reply apps/daimo-mobile/src/view/screen/SettingsScreen.tsx:331]: not sure either. i'll test on android but change lgtm

File: apps/daimo-mobile/src/view/screen/onboarding/AllowNotificationsPage.tsx (hunk 2 above)
Comment (dcposch) [reply apps/daimo-mobile/src/view/screen/onboarding/AllowNotificationsPage.tsx:53]: looks great

File: apps/daimo-mobile/src/view/shared/ContactBubble.tsx [hunk 5]
```diff
@@ -117,22 +116,25 @@ export function Bubble({
...
     () => ({
       position: "absolute",
-      height: size,
-      width: size,
-      borderRadius: size / 2,
-      zIndex: 5,
+      // Match size of bordered default bubble
+      height: size - 1,
+      width: size - 1,
+      borderRadius: 99,
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/ContactBubble.tsx:122]: `zIndex` no longer needed?

File: apps/daimo-web/src/app/download/other/page.tsx [hunk 6]
```diff
@@ -47,11 +52,20 @@ export default function DownloadPage() {
...
-            <div className="h-16" />
+            <div className="h-[350px] flex flex-row items-center justify-center">
+              <Image
+                src="/assets/download/mac.png"
+                width={240}
+                height={144}
+                alt="MacOS"
+              />
+            </div>
+            <div className="h-8" />
```
Comment (dcposch) [inline apps/daimo-web/src/app/download/other/page.tsx:68]: pics look great!

File: packages/daimo-api/src/api/profile.ts [hunk 7]
```diff
@@ -136,7 +136,9 @@ export class ProfileCache {
   }
 
   getProfilePicture(addr: Address) {
-    return `${daimoDomainAddress}/profile/${addr}/pfp`;
+    return this.getLinkedAccounts(addr)?.[0]?.pfpUrl
+      ? `${daimoDomainAddress}/profile/${addr}/pfp`
+      : undefined;
```
Comment (dcposch) [inline packages/daimo-api/src/api/profile.ts:141]: 👍

**Review by dcposch — APPROVED** [review]
Looks great!

File: apps/daimo-mobile/src/logic/notify.ts (hunk 3 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/logic/notify.ts:170]: good call yeah looks like its not -- I was just keeping it from old code, dropped.

**Pull request merged**